### ISSUE TITLE
Legend Overlaps with X-Axis Labels After Using move_legend in Seaborn

### ISSUE THREAD

**Issue by hasibagen** [mwaskom/seaborn#3721]
**Description:**

I'm encountering an issue with Seaborn's `relplot` function when using `sns.move_legend`. After moving the legend to the center-bottom of the plot, it overlaps with the x-axis labels. I've tried adjusting the subplot parameters and using `plt.rcParams['figure.autolayout']`, but the legend still overlaps with the x-axis labels. Here is a minimal reproducible example:

**Code:**

```python
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

# Generate sample data
np.random.seed(42)
time = np.linspace(0, 100, 300)
roi = ["Left Temporal", "Frontal", "Right Temporal"]
types = ["Oxyhemoglobin", "Deoxyhemoglobin", "Total Hemoglobin"]

data = []
for r in roi:
    for t in types:
        values = np.sin(time/10 + np.random.randn()) + np.random.normal(0, 0.1, len(time))
        for i in range(len(time)):
            data.append([time[i], values[i], r, t])

df_roi = pd.DataFrame(data, columns=["time", "value", "roi", "type"])

# Plotting
lineplot_device = sns.relplot(
    data=df_roi, kind="line",
    x="time", y="value", col="roi",
    hue="type",
    palette={"Oxyhemoglobin": "#FF847C", "Deoxyhemoglobin": "#82B3D0", "Total Hemoglobin": "#A2D5A2"},
    facet_kws=dict(sharex=True),
    legend="brief",
    errorbar=None,
    col_order=['Left Temporal', 'Frontal', 'Right Temporal'],
    height=3,
    aspect=0.8
)

sns.move_legend(lineplot_device, "center", bbox_to_anchor=(0.5, 0.05), ncol=3, title=None, frameon=False, numpoints=4)
lineplot_device.fig.subplots_adjust(bottom=0.25, right=1)
plt.rcParams['figure.autolayout'] = True

plt.show()
```
Expected Behavior:
The legend should be centered at the bottom of the plot without overlapping with the x-axis labels.

Actual Behavior:
The legend overlaps with the x-axis labels, making the labels difficult to read.

**Comment by mwaskom** [comment]
You’re setting `bbox_to_anchor` so you are basically asking for this?

**Comment by hasibagen** [comment]
> You’re setting `bbox_to_anchor` so you are basically asking for this?

The bbox_to_anchor function has limited adjustment. I tried all possible numbers. No matter which one, the label would overlap with the image after being placed below, or overlap with the x-axis label, or overlap with the image if bbox_to_anchor is adjusted too high, or disappear in the image after bbox_to_anchor is adjusted.
Finally, through the adjustment of bbox_to_anchor, I did not achieve the color label to be displayed below the x-axis without overlapping.
